### PR TITLE
Save File Fix

### PR DESCRIPTION
**Pull request by rjhancock**
Updates to Serialization Helper
Minor clean ups to AbstractGameManager, GameManagerSaveHelper, and Server.
Removed a nested try/catch block.

### CODE REVIEW

**Comment by rjhancock** [comment]
Should resolve #5756

File: megamek/src/megamek/server/GameManagerSaveHelper.java [hunk 1]
```diff
@@ -95,8 +108,10 @@ public void sendSaveGame(int connId, String fileName, String localPath) {
                 finalFileName = fileName + MMConstants.SAVE_FILE_GZ_EXT;
             }
         }
-        localPath = localPath.replaceAll("\\|", " ");
+
+        localPath = localPath.replace("|", " ");
```
Comment (Sleet01) [inline megamek/src/megamek/server/GameManagerSaveHelper.java:112]: Was the escaping of the "|" actually unnecessary here?

Comment (rjhancock) [reply megamek/src/megamek/server/GameManagerSaveHelper.java:112]: `replaceAll` defaults to a regex pattern and create unneeded overhead if it's just a string.
`replace` defaults to string and only moves to regex when needed.

**Review by Sleet01 — APPROVED** [review]
LGTM

**Pull request merged**